cas-server-support-oauth/src/main/java/org/jasig/cas/config/OAuthConfiguration.java

```diff
@@ -51,14 +73,35 @@
     public Config oauthSecConfig() {
         final CasClient oauthCasClient = new CasClient(this.casLoginUrl);
         oauthCasClient.setName(CAS_OAUTH_CLIENT);
+        oauthCasClient.setCallbackUrlResolver((url, context) -> {
```

Comment:
Dont think so. We'll end up then doing double filtering, because the authentication principal that is created will contain all attributes and we would then have to filter them out based on what the actual RP configuration is, which also means doubly calls to the service registry. We also wont be able to do the filtering at all for the profile requests, since we wont know what the service is. All we have is just an access token and authentication. 

You always want to pick up the rules for attributes for the *real* RP. Callback is a fake url, and shouldnt have to have any kind of special logic.